automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/Leonardo.scala

```diff
@@ -147,27 +148,32 @@ object Leonardo extends RestClient with LazyLogging {
       mapper.readValue(response, classOf[ContentItem])
     }
 
-    def notebooksPath(googleProject: GoogleProject, clusterName: ClusterName): String =
+    def notebooksBasePath(googleProject: GoogleProject, clusterName: ClusterName): String =
       s"notebooks/${googleProject.value}/${clusterName.string}"
 
     def contentsPath(googleProject: GoogleProject, clusterName: ClusterName, contentPath: String): String =
-      s"${notebooksPath(googleProject, clusterName)}/api/contents/$contentPath"
+      s"${notebooksBasePath(googleProject, clusterName)}/api/contents/$contentPath"
 
     def localizePath(googleProject: GoogleProject, clusterName: ClusterName, async: Boolean = false): String =
-      s"${notebooksPath(googleProject, clusterName)}/api/localize${if (async) "?async=true" else ""}"
+      s"${notebooksBasePath(googleProject, clusterName)}/api/localize${if (async) "?async=true" else ""}"
 
     def get(googleProject: GoogleProject, clusterName: ClusterName)(implicit token: AuthToken, webDriver: WebDriver): NotebooksListPage = {
-      val path = notebooksPath(googleProject, clusterName)
+      val path = notebooksBasePath(googleProject, clusterName)
       logger.info(s"Get notebook: GET /$path")
       new NotebooksListPage(url + path)
     }
 
     def getApi(googleProject: GoogleProject, clusterName: ClusterName)(implicit token: AuthToken): String = {
-      val path = notebooksPath(googleProject, clusterName)
+      val path = notebooksBasePath(googleProject, clusterName)
       logger.info(s"Get notebook: GET /$path")
       parseResponse(getRequest(url + path))
     }
 
+    def getApiHeaders(notebookUrl: String)(implicit token: AuthToken): Seq[HttpHeader] = {
```

Comment:
All the other methods in this class build the URL from a (GoogleProject, ClusterName). Seems like this one could do similar.